File: backend/app/schemas/triage.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from uuid import UUID
from datetime import datetime
from .enums import TriageSessionStatus, UrgencyLevel, EscalationType, ChatRetentionPolicy, MessageSender, MessageType, AgentPhase


class TriageSessionCreate(BaseModel):
    chief_complaint: Optional[str] = None


class TriageSessionResponse(BaseModel):
    """Full session response — used by doctor-facing endpoints."""
    id: UUID
    patient_id: UUID
    doctor_id: UUID
    hospital_id: UUID
    status: TriageSessionStatus
    chief_complaint: Optional[str]
    detected_symptoms: Optional[Dict[str, Any]]
    urgency_level: UrgencyLevel
    escalation_type: EscalationType
    chat_retention_policy: ChatRetentionPolicy
    agent_phase: Optional[AgentPhase]
    started_at: datetime
    ended_at: Optional[datetime]

    class Config:
        from_attributes = True


class PatientTriageSessionResponse(BaseModel):
    """Patient-safe session response — excludes urgency, symptoms, and diagnosis data."""
    id: UUID
    status: TriageSessionStatus
    chief_complaint: Optional[str]
    agent_phase: Optional[AgentPhase]
    started_at: datetime
    ended_at: Optional[datetime]

    class Config:
        from_attributes = True


class SessionPhaseResponse(BaseModel):
    """Current agent phase for the session."""
    phase: Optional[AgentPhase]
    is_escalated: bool = False

    class Config:
        from_attributes = True


class MessageCreate(BaseModel):
    content: str


class MessageResponse(BaseModel):
    id: UUID
    session_id: UUID
    sender: MessageSender
    content: str
    message_type: MessageType
    is_persisted_after_summary: bool
    is_visible_to_doctor: bool
    is_deleted: bool
    created_at: datetime

    class Config:
        from_attributes = True


class ClinicalReportResponse(BaseModel):
    id: UUID
    session_id: UUID
    patient_id: UUID
    doctor_id: UUID
    presenting_complaints: Optional[Any]
    history_of_presenting_complaint: Optional[Any]
    summary_text: str
    suspected_conditions: Optional[Any]
    triggered_red_flags: Optional[Any]
    urgency_level: UrgencyLevel
    recommended_action: str
    specialty_routing: Optional[Any]
    suggested_workup: Optional[Any]
    key_exam_findings: Optional[Any]
    admission_criteria: Optional[Any]
    referral_criteria: Optional[Any]
    external_escalation_completed: bool
    escalation_message: Optional[str]
    visible_to_patient: bool
    model_version: Optional[str]
    diagnosis_mode: Optional[str]
    diagnosis_pass_count: Optional[int]
    chunks_used_count: Optional[int]
    web_search_results: Optional[Any]
    generated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/triage.ts
import {
  TriageSessionStatus,
  UrgencyLevel,
  EscalationType,
  ChatRetentionPolicy,
  MessageSender,
  MessageType,
  AgentPhase,
} from './enums';

export interface TriageSession {
  id: string;
  patient_id: string;
  doctor_id: string;
  hospital_id: string;
  status: TriageSessionStatus;
  chief_complaint: string | null;
  detected_symptoms: Record<string, unknown> | null;
  urgency_level: UrgencyLevel;
  escalation_type: EscalationType;
  chat_retention_policy: ChatRetentionPolicy;
  agent_phase: AgentPhase | null;
  started_at: string;
  ended_at: string | null;
}

/**
 * Patient-safe session response — excludes urgency, symptoms, and diagnosis data.
 * This is what patient-facing endpoints return.
 */
export interface PatientTriageSession {
  id: string;
  status: TriageSessionStatus;
  chief_complaint: string | null;
  agent_phase: AgentPhase | null;
  started_at: string;
  ended_at: string | null;
}

export interface TriageSessionCreate {
  chief_complaint?: string | null;
}

export interface Message {
  id: string;
  session_id: string;
  sender: MessageSender;
  content: string;
  message_type: MessageType;
  is_persisted_after_summary: boolean;
  is_visible_to_doctor: boolean;
  is_deleted: boolean;
  created_at: string;
}

export interface MessageCreate {
  content: string;
}

export interface ClinicalReport {
  id: string;
  session_id: string;
  patient_id: string;
  doctor_id: string;
  presenting_complaints: unknown;
  history_of_presenting_complaint: unknown;
  summary_text: string;
  suspected_conditions: unknown;
  triggered_red_flags: unknown;
  urgency_level: UrgencyLevel;
  recommended_action: string;
  specialty_routing: unknown;
  suggested_workup: unknown;
  key_exam_findings: unknown;
  admission_criteria: unknown;
  referral_criteria: unknown;
  external_escalation_completed: boolean;
  escalation_message: string | null;
  visible_to_patient: boolean;
  model_version: string | null;
  diagnosis_mode: string | null;
  diagnosis_pass_count: number | null;
  chunks_used_count: number | null;
  generated_at: string;
}

/**
 * Session phase response from GET /triage/sessions/{id}/phase
 */
export interface SessionPhase {
  phase: AgentPhase | null;
  is_escalated: boolean;
}

/**
 * Feedback case from the FeedbackStore (doctor-facing).
 */
export interface FeedbackCase {
  case_id: string;
  patient_name: string;
  timestamp: string;
  symptoms: string[];
  urgency: string;
  intake_summary: Record<string, unknown>;
  clinical_picture: Record<string, unknown>;
  diagnosis_report: Record<string, unknown>;
  system_primary_diagnosis: string;
  review_status: 'pending' | 'confirmed' | 'rejected';
  doctor_decision: string | null;
  doctor_diagnosis: string | null;
  doctor_notes: string | null;
}

/**
 * Feedback statistics from GET /doctor/feedback/statistics
 */
export interface FeedbackStats {
  total_cases: number;
  confirmed: number;
  rejected: number;
  pending: number;
}
